modules/11_subclustering/subclustering.qmd

```diff
@@ -0,0 +1,676 @@
+# In this case, when using on-disk matrices, they need to be written to disk again
+barcodes_removed = FALSE
+```
+
+<!-- 
+## Remove cells
+
+USER_INPUT:
+
+- Remove the HTML comments
+- Set eval in the chunk options to true
+- Add explanation why barcodes are remove here
+
+```{r}
+#| label: subclustering_remove_barcodes_1
+#| results: asis
+#| eval: false
+
+# Use this chunk to remove barcodes from the analysis.
+# However note that it might be better to remove these barcodes already in the QC module.
+
+# USER INPUT: 
+# - Specify which barcodes of the default assay to remove
+bcs = SeuratObject::Cells(sc[[default_assay]])
+barcode_metadata = sc[[]][bcs, ]
+
+barcodes_to_remove = barcode_metadata %>%
+  dplyr::filter(seurat_clusters == "1") %>%
+  rownames()
+
+# Filter barcodes:
+# a) barcodes that are present only in this assay, will be removed from the whole Seurat object
+# b) barcodes that are present in other assays, will be removed from this assay only
+barcodes_other_assays = purrr::map(setdiff(SeuratObject::Assays(sc), default_assay), function(a) return(SeuratObject::Cells(sc[[a]]))) %>% 
+  purrr::flatten_chr()
+
+# Remove from assay barcodes that are still present in other assays
+barcodes_to_remove_assay = intersect(barcodes_to_remove, barcodes_other_assays)
+barcodes_to_keep = setdiff(SeuratObject::Cells(sc[[default_assay]]), barcodes_to_remove_assay)
+sc[[default_assay]] = subset(sc[[default_assay]], cells=barcodes_to_keep)
+
+# Remove from Seurat object barcodes that are not present in other assays
+barcodes_to_remove_sc = setdiff(barcodes_to_remove, barcodes_other_assays)
+if (length(barcodes_to_remove_assay) > 0) {
+  barcodes_sc = purrr::map(SeuratObject::Assays(sc), function(a) return(SeuratObject::Cells(sc[[a]]))) %>% purrr::flatten_chr() %>% unique()
+  barcodes_sc = setdiff(barcodes_sc, barcodes_to_remove_sc)
+  sc = subset(sc, cells=barcodes_sc)
+}
+
+# Indicate that barcodes were removed so that on-disk matrices (if used) are updated
+barcodes_removed = TRUE
+
+# Print information
+CalloutBox(x=FormatString("Removed {length(barcodes_to_remove)} barcodes from the analysis.", quote=FALSE), type="note")
+```
+
+```{r}
+#| label: fig-subclustering_remove_barcodes_2
+#| fig-cap: 'Clusters after removing barcodes'
+#| eval: false
+
+# Plot a UMAP
+DimPlot(sc, label=TRUE) + 
+  AddPlotStyle()
+```
+
+-->
+
+<!--
+## Merge/rename cluster
+
+USER_INPUT:
+
+- Remove the HTML comments
+- Set eval in the chunk options to true
+- Add explanation why clusters are merged/renamed here
+
+```{r}
+#| label: subclustering_merge_rename_cluster_1
+#| tbl-cap: 'Clusters after merging and/or renaming'
+#| eval: false
+
+# Use this chunk to merge or rename clusters in the Seurat object
+
+# Get barcodes for default assay
+bcs = SeuratObject::Cells(sc[[default_assay]])
+
+# USER_INPUT: 
+# - Set cluster column and what clusters to merge/rename into a new column 'merged_clusters'
```

Comment:
Can you give an example in the comments? It's not immediately clear to me what I have to do with `sc$merged_clusters`.